p2p/net/swarm/swarm.go

```diff
@@ -108,6 +108,28 @@ func WithDialRanker(d network.DialRanker) Option {
 	}
 }
 
+// WithUDPBlackHoleConfig configures swarm to use c as the config for UDP black hole detection
+func WithUDPBlackHoleConfig(c *BlackHoleConfig) Option {
```

Comment:
Alternatively, you could just make these 3 function parameters:
```go
func WithUDPBlackHoleConfig(enabled bool, n, min int)
```
and then make `BlackHoleConfig` an unexported struct.